File: 백준/Bronze/5585. 거스름돈/거스름돈.py
import sys
input = sys.stdin.readline

N = int(input())
v = 1000 - N

coins = [500,100,50,10,5,1]
res = 0
idx = 0

while True:
    if v == 0:
        break

    cnt = v // coins[idx]

    if cnt:
        res += cnt

    v -= cnt * coins[idx]
    idx += 1

print(res)

File: goormlevel/175177/1. 거스름돈/거스름돈.js
const solve = (N) => {
	let won = N;
	let rtn = 0;
	let cnt;
	for (let coin of [40, 20, 10, 5, 1]){
		cnt =  Math.floor(won / coin);
		rtn += cnt;
		won -= (cnt * coin);
	}
	return rtn;
}

// Run by Node.js
const readline = require('readline');

(async () => {
	let rl = readline.createInterface({ input: process.stdin });
	
	for await (const line of rl) {
		console.log(solve(parseInt(line)));
		rl.close();
	}
	
	process.exit();
})();
